src/Process/3dTilesProcessing.js

```diff
@@ -236,20 +236,27 @@ export function pre3dTilesUpdate(context, layer) {
     const HYFOV = 2.0 * Math.atan(Math.tan(radAngle * 0.5) * hypotenuse / context.camera.width);
     context.camera.preSSE = hypotenuse * (2.0 * Math.tan(HYFOV * 0.5));
 
-    // once in a while, garbage collect
-    if (Math.random() > 0.98) {
+    // Elements removed are added in the layer._cleanableTiles list.
+    // Since we simply push in this array, the first item is always
+    // the oldest one.
+    const now = Date.now();
+    if (layer._cleanableTiles.length
+        && (now - layer._cleanableTiles[0].cleanableSince) > layer.cleanupDelay) {
         // Make sure we don't clean root tile
         layer.root.cleanableSince = undefined;
 
-        // Browse
-        const now = Date.now();
-
-        for (const elt of layer._cleanableTiles) {
+        let i = 0;
+        for (; i < layer._cleanableTiles.length; i++) {
+            const elt = layer._cleanableTiles[i];
```

Comment:
Since I think you are going for optimization here, maybe move the `const` declaration outside the loop ?

Reply:
- this PR is not really about optimizing - I moved away from the `for .. of` loop because I need the index anyway (cf `layer._cleanableTiles.splice(0, i);` after)
- the `const elet = ...` depends on the loop indice, so I cannot move it outside

Reply:
Ok !